File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices)
    {
        int minPrice = prices[0];
        int maxProfit = 0;
        for (const int &p : prices)
        {
            if (p < minPrice)
            {
                minPrice = p;
            }
            else
            {
                maxProfit = max(maxProfit, (p - minPrice));
            }
        }
        return maxProfit;
    }

};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s == "":
            return 0
        l, r = 0, 0
        chars = set()
        maxL = 0
        while r < len(s):
            if s[r] not in chars:
                chars.add(s[r])
                maxL = max(maxL, (r-l+1))
                r += 1
            elif chars:
                chars.remove(s[l])
                l += 1
        return maxL